src/drivers/uavcan_v1/templates/_composite_type.j2

```diff
@@ -0,0 +1,115 @@
+{{ composite_type | definition_begin }}
+{
+{%- if T.fixed_port_id is not None %}
+	static constexpr CanardPortID PORT_ID = {{T.fixed_port_id}};
+{%- endif %}
+
+{%- set total_size = namespace(value=0) -%}
+{%- for field in composite_type.fields -%}
+    {% set total_size.value = total_size.value + field.data_type.bit_length_set|max %}
+{%- endfor %}
+	static constexpr size_t SIZE = {{total_size.value}};
+
+	static constexpr auto getDataTypeFullName() { return "{{T}}"; }
+
+{% for constant in composite_type.constants %}
+	static constexpr {{ constant.data_type | declaration }} {{ constant.name | id }} = {{ constant.value.native_value.numerator }} / {{ constant.value.native_value.denominator }};
+{%- endfor -%}
+
+{% if composite_type is UnionType %}
+#error "TODO: UnionType
+{% else %}
+
+{% for field in composite_type.fields -%}
+    {%- if field is not padding %}
+        {%- if field.data_type is FloatType %}
+            {%- if field.data_type.bit_length_set|max <= 32 %}
+	float {{ field | id }}{NAN};
+            {%- else %}
+	double {{ field | id }}{NAN};
+            {%- endif %}
+        {%- elif field.data_type is BooleanType %}
+	bool {{ field | id }}{false};
+        {%- else %}
+	{{ field.data_type | declaration }} {{ field | id }}{};
+        {%- endif -%}
+    {%- endif -%}
+{%- endfor %}
+{% endif %}
+
+	void serializeToBuffer(uint8_t* const buffer, const size_t starting_bit = 0)
+	{
+{%- set bit_offset = namespace(value=0) -%}
+{%- for field in composite_type.fields -%}
+    {%- if field is not padding %}
+      {%- if field.data_type is SerializableType %}
+        {%- if field.data_type is IntegerType %}
+		canardDSDLSetUxx(buffer, starting_bit + {{ bit_offset.value }}, {{ field.name }}, {{ field.data_type.bit_length_set | max }});
+        {%- elif field.data_type is BooleanType %}
+		canardDSDLSetBit(buffer, starting_bit + {{ field.data_type.bit_length_set | max }}, {{ field.name }});
```

Comment:
Wrong offset calculation should be `bit_offset.value `instead of `field.data_type.bit_length_set | max`

Reply:
Thanks Peter!

Reply:
(for primitives there is also a shortcut [`bit_length: int`](https://github.com/UAVCAN/pydsdl/blob/f53eb68f7c403fe27db5f8ff18e45289b6e14cae/pydsdl/_serializable/_primitive.py#L50-L53))